extensions/test/algorithms/parse.cpp

```diff
@@ -2,36 +2,38 @@
 // Unit Test
 
 // Copyright (c) 2021 Joni Kerkel�, Oulu, Finland.
+// Copyright (c) 2021 Ayush Gupta, Gujarat, India
 
 // Use, modification and distribution is subject to the Boost Software License,
 // Version 1.0. (See accompanying file LICENSE_1_0.txt or copy at
 // http://www.boost.org/LICENSE_1_0.txt)
 
+#include <string>
 #include <geometry_test_common.hpp>
 
 #include <boost/geometry/core/coordinate_type.hpp>
 #include <boost/geometry/extensions/algorithms/parse.hpp>
 #include <boost/geometry/geometries/point.hpp>
 #include <boost/geometry/util/math.hpp>
-#include <boost/geometry/srs/projections/impl/dms_parser.hpp>
 
 #include <test_common/test_point.hpp>
 
-
 using namespace boost::geometry;
 
-template <typename P>
-void test_parse(std::string const& coordinate_1, std::string const& coordinate_2,
-	double expected_result_coord_1, double expected_result_coord_2)
+template <typename Geometry, typename T>
+void test_parse(std::string const& coordinate_1,
+	std::string const& coordinate_2,
+	T expected_result_coord_1,
+	T expected_result_coord_2)
 {
-	auto geo_point_result = parse<P>(coordinate_1, coordinate_2);
-	BOOST_CHECK_CLOSE(((double)bg::get<0>(geo_point_result)), expected_result_coord_1, 1.0e-6);
-	BOOST_CHECK_CLOSE(((double)bg::get<1>(geo_point_result)), expected_result_coord_2, 1.0e-6);
-
-	P p;
-	parse(p, coordinate_1, coordinate_2);
-	BOOST_CHECK_CLOSE(((double)bg::get<0>(p)), expected_result_coord_1, 1.0e-6);
-	BOOST_CHECK_CLOSE(((double)bg::get<1>(p)), expected_result_coord_2, 1.0e-6);
+	Geometry geometry;
+	geometry = parse<Geometry>(coordinate_1, coordinate_2);
+	BOOST_CHECK_CLOSE(((double)bg::get<0>(geometry)), expected_result_coord_1, 1.0e-4);
+	BOOST_CHECK_CLOSE(((double)bg::get<1>(geometry)), expected_result_coord_2, 1.0e-4);
```

Comment:
Such change from `1.0e-6` to `1.0e-4`  needs a word of comment, ideally, in commit message

Reply:
mentioned now

Reply:
Yes, the commit https://github.com/boostorg/geometry/pull/806/commits/138d3d53e66a65dfd12be0af7815bc9493ed49fc  does mention it:

```
and reduce the error range from 1e-6 to 1e-4
```

but this is just *repeating **what** the commit does change* (i.e. it is equivalent to what everyone can see in the diff :-)), but it does not explain **why** the range changed. The **why** is what the commit messages are for. The **what** everyone can see by looking at the diff of a commit.

So, I'm hitting "Unresolve conversation" as this still needs clarification.

Reply:
The range changed because I have also included test cases where the final value is very close to zero, so even though the difference between the actual value and expected value is of order 1e-9 , the percentage error exceed the previous error range of 1e-6 and had to be reduced to 1e-4. This was the case when 10 precision points were specified in the expected value. Is this explanation satisfying enough? If so I will add the explanation in the commit message after your confirmation.

Reply:
Yes, it is indeed.